packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -2959,6 +2959,23 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     });
   }
 
+  @override
+  void handleKeyboardUndo(String direction) {
+    if (_editableKey.currentContext == null)
+        return;
+    if (direction == 'undo') {
+      const UndoTextIntent intent = UndoTextIntent(SelectionChangedCause.keyboard);
+      final Action<UndoTextIntent>? action = Actions.maybeFind<UndoTextIntent>(_editableKey.currentContext!, intent: intent);
+      if (action != null)
```

Comment:
this can be moved out of the both if(direction == 'undo') and else case so that we don't need to duplicate code